### PR TITLE
irreversible admin tasks

### PR DESCRIPTION
**Pull request by xithan**
and fix dirty attributes handling

### CODE REVIEW

File: card/mod/admin/spec/set/self/admin_spec.rb [hunk 1]
```diff
@@ -9,45 +9,62 @@
...
         expect(Card.where(trash: true)).not_to be_empty
-        Card::Env.params[:task] = :empty_trash
-        @admin.update_attributes({})
+        run_admin_task :empty_trash
         expect(Card.where(trash: true)).to be_empty
       end
-    end
 
-    # NOTE: I removed this functionality for now, because I don't think we
-    # should have web access to admin functions that can incur actual data loss.
```
Comment (xithan) [inline card/mod/admin/spec/set/self/admin_spec.rb]: You only removed the method but not the link. I brought the method back but disabled it by default with an error message that explains how to turn it on.

File: card/mod/history/set/all/content_history.rb [hunk 2]
```diff
@@ -166,8 +168,7 @@ def draft_acts
   drafts.created_by(Card::Auth.current_id).map(&:act)
 end
 
-event :detect_conflict, :validate, on: :update,
-                                   when: proc { |c| c.edit_conflict? } do
+event :detect_conflict, :validate, on: :update, when: proc { |c| c.edit_conflict? } do
```
Comment (ethn) [inline card/mod/history/set/all/content_history.rb]: I think `when: :edit_conflict?` should work.

Comment (ethn) [reply card/mod/history/set/all/content_history.rb]: (I mean, I think it already works)

File: decko/lib/decko/config/initializers/sedate_parser.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Hack to get rid of annoying parser warnings
```
Comment (ethn) [inline decko/lib/decko/config/initializers/sedate_parser.rb]: nicely named.

File: card/mod/carrierwave/lib/carrier_wave/file_card_uploader.rb [hunk 4]
```diff
@@ -264,7 +261,7 @@ def original_filename
     end
 
     def action_id
-      model.selected_content_action_id || NO_ACTION_ID
```
[the remark this answers is not in the document]
Comment (xithan) [reply card/mod/carrierwave/lib/carrier_wave/file_card_uploader.rb]: no, it caused problems when I turned off the history for machine output. The generate machine output had no longer a file name.

File: card/mod/history/set/all/content_history.rb [hunk 5]
```diff
@@ -94,25 +98,23 @@ def new_content_action_id
 end
 
 def current_action_changes_content?
-  new_card? || @current_action.new_content? || db_content_changed?
+  new_card? || @current_action.new_content? || db_content_is_changing?
 end
 
 def last_action_id
-  (la = last_action) && la.id
+  last_action&.id
```
[the remark this answers is not in the document]
Comment (xithan) [reply card/mod/history/set/all/content_history.rb]: That's safe to assume. It's the same as `last_action.id`. The only difference is that it won't fail if last_action is nil. Instead it returns nil.

**Pull request merged**